tests/data/ref_dim_keyword.nxdl.xml

```diff
@@ -30,34 +30,52 @@
             <doc>
                  Test
             </doc>
-            <dimensions>
+            <dimensions rank="1">
                 <dim index="1" value="2"/>
             </dimensions>
         </field>
         <field name="my_dim_array2" type="NX_NUMBER">
             <doc>
                  Testarray with dimensions spelled out
             </doc>
-            <dimensions rank="1">
+            <dimensions rank="2">
                 <dim index="1" value="10"/>
                 <dim index="2" value="30"/>
             </dimensions>
         </field>
         <field name="my_dim_with_numpy_3d" type="NX_NUMBER">
-            <dimensions>
+            <dimensions rank="3">
                 <dim index="1" value="2"/>
                 <dim index="2" value="4"/>
                 <dim index="3" value="5"/>
             </dimensions>
         </field>
-        <field name="my_dim_with_doc_and_dim_attr" type="NX_NUMBER">
-            <dimensions>
+        <field name="my_dim_with_doc_rank_and_dim_attr" type="NX_NUMBER">
+            <dimensions rank="2">
                 <doc>
                      Test with doc and dim attribute
                 </doc>
                 <dim index="1" value="2" ref="s"/>
                 <dim index="3" value="4" ref="m"/>
             </dimensions>
         </field>
+        <field name="my_dim_with_doc_and_short_dim_attr" type="NX_NUMBER">
+            <dimensions>
```

Comment:
We must have a `rank` attribute here other in `dimenstion` XML element as well if the rank is not mentioned in `yaml`.

Reply:
You are right, it should write the rank here, so that should actually be added to the test here.

Currently, however, the tool does not write the rank here.

I have added it to the reference `ref_dim_keyword.nxdl.xml` file, so that when the tool is working properly, the tests are passed.